### ISSUE TITLE
网页版chatgpt无法使用

### ISSUE THREAD

**Issue by MrX1983TW** [josStorer/chatGPTBox#469]
现在点选网页版的ChatGPT都会出现以下讯息：
错误:

 需要通过 OpenAI 的安全检查
 请开启 https://chat.openai.com
 接着点击右上角的「重试」按钮

 建议在 https://platform.openai.com/account/api-keys 建立一个 API 金钥

开启网页登入后仍然无法使用(我有订阅)，请再协助修正

**Comment by josStorer** [comment]
你使用的模式是什么，GPT-4吗，试试Mobile模式

**Comment by MrX1983TW** [comment]
对，是GPT 4，但是切换到Mobile模式首先会出现下面讯息：{"detail":"Only one message at a time. Please allow any other responses to complete before sending another message, or wait one minute."}
过一分钟再试还是会出现上面的错误要我重新登入

**Comment by josStorer** [comment]
这是你刚才发送了别的请求，点击重试按钮

**Comment by MrX1983TW** [comment]
![Screenshot_2023-07-20-17-23-28-26_ffb2f5e1b976ff98cfc94f359fbce8de](https://github.com/josStorer/chatGPTBox/assets/61070791/d1bb45bb-41be-411e-9916-bdb0ba1fdae3)
还是不行，会变成上面那样

**Comment by josStorer** [comment]
试试保持chatgpt官网打开使用，点击图片里的链接跳转到官网，然后不要关闭

**Comment by MrX1983TW** [comment]
还是不行，刚刚用电脑版的chrome重试也是一样

**Comment by MrX1983TW** [comment]
如果不选GPT 4，则网页版没有问题，但是使用GPT 4就会出现上面的错误

**Comment by josStorer** [comment]
是2.4.0版本吗

**Comment by MrX1983TW** [comment]
是2.4.0版本没错

**Comment by betterdoitnow** [comment]
我也有同样的问题.... 只要选了GPT-4模型，就是报错

**Comment by lim142857** [comment]
It appears that GPT-4 version is still not functioning.

**Comment by xuuou** [comment]
我也是同样的问题，选择GPT-4就会报错（已经订阅plus）。

**Comment by yanzhengdao** [comment]
同样问题，无法选择GPT-4, plus用户，现在plus已经默认GPT-4了，请更新

**Comment by josStorer** [comment]
目前GPT4的Web请求依赖arkose_token进行验证，这个token的获取在浏览器扩展的客户端侧不太好实现，POE也是类似的情况

**Comment by josStorer** [comment]
chathub自建了服务器，获取arkose_token，并且由于它的代码是开源的，理论上我可以向它的服务器请求获取arkose_token，但我觉得这么做有些不道德

**Comment by josStorer** [comment]
@lethefrost 暂时没找到解决方法，找到了会更新
https://github.com/josStorer/chatGPTBox/releases/tag/v2.4.1

**Comment by Dylan-Li8** [comment]
如果可以私有化部署服务器端项目用以支持获取arkose_token，这样是否可行？

**Comment by Dylan-Li8** [comment]
或者，能否使用https://chat.zhile.io/的接口来实现？

**Comment by Dylan-Li8** [comment]
另外，请查看一下maxai.me 是怎么实现的，它是可以在浏览器扩展里使用GPT-4网页版。

**Comment by josStorer** [comment]
@Dylan-Li8 私有化部署是可行的，这个项目支持自定义API和 [waylaidwanderer](https://github.com/waylaidwanderer/node-chatgpt-api) API，你可以随意编写一个本地http服务器，返回的json响应中的response字段会自动使用，或者你也可以自己实现openai API兼容的sse响应，同样支持

**Comment by PeterDaveHello** [comment]
Not sure if it helps, looks like bravegpt is working properly: https://github.com/KudoAI/bravegpt

**Comment by josStorer** [comment]
@PeterDaveHello Thanks, i'll take a look later

**Comment by levie-vans** [comment]
我看到有两个浏览器扩展实现了网页使用GPT-4：
1、HARPA AI https://chrome.google.com/webstore/detail/harpa- ai-automation-agent/eanggfilgoajaocelnaflolkadkeghjp?hl=en-GB&authuser=1
2、MaxAI.me https://chrome.google.com/webstore/detail/maxaime-use-chatgpt-ai-an/mhnlakgilnojmhinhkckjpncpbhabphi

供参考

**Comment by Dylan-Li8** [comment]
我看了一下HARPA，确实是纯客户端实现的获取arkose_token，用的是OpenAI自己的js库：
```
https://tcr9i.chat.openai.com/cdn/fc/js/6af2c0d87b9879cbf3365be1a208293f84d37b1e/standard/funcaptcha_api.js?onload=loadChallenge
https://tcr9i.chat.openai.com/cdn/fc/assets/ec-game-core/bootstrap/1.15.0/standard/game_core_bootstrap.js
```

![image](https://github.com/josStorer/chatGPTBox/assets/128008286/f7c678c8-9ba0-4969-aab6-b16940a75d03)


> I came across two browser extensions that utilize the GPT-4 interface from the web-based ChatGPT, offering the complete capabilities of GPT-4:
> 
> 1. **HARPA AI**: You can check out HARPA AI at [this link](https://chrome.google.com/webstore/detail/harpa-ai-automation-agent/eanggfilgoajaocelnaflolkadkeghjp?hl=en-GB&authuser=1).
> 2. **MaxAI.me**: Another option is MaxAI.me, which you can find [here](https://chrome.google.com/webstore/detail/maxaime-use-chatgpt-ai-an/mhnlakgilnojmhinhkckjpncpbhabphi).
> 
> Feel free to take a look!

**Comment by josStorer** [comment]
Fixed in https://github.com/josStorer/chatGPTBox/commit/4859e9a905e8a405d2a5c40d526bec1b2ea112ed
Thank you for providing these references. In the end, I adopted a method that I believe may be more reliable to obtain the arkose_token. This method does not depend on the specific api.js from the official repository and script injection, making it usable on any page.

**Comment by duAffentier** [comment]
Update for firefox?

**Comment by josStorer** [comment]
@duAffentier Waiting for review
![image](https://github.com/josStorer/chatGPTBox/assets/13366013/08f7374e-3cde-4c8f-b0ec-4fe91e86be04)

**Comment by Aerasyn** [comment]
Yes, but the update is not being downloadable in firefox extensiion store?

**Comment by PeterDaveHello** [comment]
Same here, that's strange...

> Yes, but the update is not being downloadable in firefox extension store? This is all I get so far: ![image](https://user-images.githubusercontent.com/933664/280069589-9ab2910b-4f3d-47c9-9ca3-3b0522a669b6.png)

**Comment by josStorer** [comment]
@Aerasyn @PeterDaveHello 
I am dealing with an audit issue with Firefox. Firefox believes that the compiled output of the source code I submitted does not match the contents of the submitted compressed file. As a result, the addon has been temporarily disabled. I think it will not take too long to resolve this issue.

Regarding the zip file, did you download it from https://github.com/josStorer/chatGPTBox/releases/download/v2.4.4/firefox.zip ? It works fine on my computer. Please refer to the instructions here for installation: https://github.com/josStorer/chatGPTBox/wiki/Install#install-to-firefox.

**Comment by PeterDaveHello** [comment]
The latest version is now available :tada: 

![image](https://github.com/josStorer/chatGPTBox/assets/3691490/27735f9c-2dac-4b98-904e-3e11fb9d66ae)

**Comment by endolith** [comment]
This Add-on doesn't work in Firefox, version 2.5.5:

> Error:
> 
> OpenAI Security Check Required
> Please open https://chatgpt.com
> And click the retry button in the top right corner
> 
> Consider creating an api key at https://platform.openai.com/account/api-keys

**Comment by endolith** [comment]
Oh I found it.  You need to change default model first, and then API key box appears:

![image](https://github.com/josStorer/chatGPTBox/assets/58611/47d3a1c8-d02f-491b-82e8-5f8ca48f1b13)
